changelog.d/10963.bugfix

```diff
@@ -0,0 +1 @@
+For the `create_room_as` test util, change the behaviour of the `is_public` option to explicitly set the `visibility` parameter of room creation requests to `true`, or to let the server decide on what value to use if left as `None`. Contributed by @AndrewFerr.
```

Comment:
There's some guidance somewhere about writing changelog entries. In particular I think we recommend they start with a verb. How does something like this sound?

```suggestion
Fix the test utility function `create_room_as` so that `is_public=True` will explicitly set the `visibility` parameter of room creation requests to `true`. Contributed by @AndrewFerr.
```

Reply:
I think this should be in a file called `10963.misc`, because `.bugfix` refers to a user-visibile bug in the application itself; here we have a bugfix in the test setup.